src/sourcekit-lsp/uriConverters.ts

```diff
@@ -0,0 +1,63 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VS Code Swift open source project
+//
+// Copyright (c) 2024 the VS Code Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VS Code Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from "vscode";
+
+// Regular expression to decompose a URI, copied from
+// https://github.com/microsoft/vscode/blob/1647dde06cd9b610e2576f09f68101c31c00e574/src/vs/base/common/uri.ts#L80
+const _decodeUriRegexp = /^(([^:/?#]+?):)?(\/\/([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?/;
```

Comment:
This works, but is it possible to use Node’s [URL](https://nodejs.org/api/url.html#new-urlinput-base) implementation here over the regex? All the sourcekit lsp Uris should be compatible with the initializer.

You’d have to build the authority out of the urls host and optional port.

Reply:
Internally VSCode uses decodeURIComponent

Reply:
Good suggestion with using Node’s URL type to do the parsing. I updated the PR to use Node’s URL parser.